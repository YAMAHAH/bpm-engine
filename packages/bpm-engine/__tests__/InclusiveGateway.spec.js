import fs from 'fs';

import BPMEngine from 'bpm-engine';

import History from './Plugins/History';
import sleep from './Plugins/sleep';

describe('InclusiveGateway', () => {
  it('Continues on all paths', async () => {
    const history = new History();
    const bpm = new BPMEngine({
      plugins: [history],
    });

    const token = await bpm.createProcessInstance({
      workflowDefinition: fs.readFileSync(`${__dirname}/diagrams/InclusiveGateway.bpmn`, 'utf-8'),
      payload: {
        top: true,
        bottom: true,
      },
    });

    await token.execute();
    await sleep(100);
    expect(history.store).toMatchSnapshot();
  });

  it('Continues on one path', async () => {
    const history = new History();
    const bpm = new BPMEngine({
      plugins: [history],
    });

    const token = await bpm.createProcessInstance({
      workflowDefinition: fs.readFileSync(`${__dirname}/diagrams/InclusiveGateway.bpmn`, 'utf-8'),
      payload: {
        top: true,
      },
    });

    await token.execute();
    await sleep(100);
    expect(history.store).toMatchSnapshot();
  });
});
