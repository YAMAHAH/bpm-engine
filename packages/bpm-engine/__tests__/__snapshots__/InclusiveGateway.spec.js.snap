// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InclusiveGateway Continues on all paths 1`] = `
Array [
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "InclusiveGateway_1sopzs7",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "InclusiveGateway_1sopzs7",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "InclusiveGateway_1sopzs7",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_0j4pnbs",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_0j4pnbs",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_0j4pnbs",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "InclusiveGateway_1uhxkjz",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_0t7t9b7",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_0t7t9b7",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_0t7t9b7",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "InclusiveGateway_1uhxkjz",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "InclusiveGateway_1uhxkjz",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "EndEvent_05zfgny",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "EndEvent_05zfgny",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "EndEvent_05zfgny",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
]
`;

exports[`InclusiveGateway Continues on one path 1`] = `
Array [
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "InclusiveGateway_1sopzs7",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "InclusiveGateway_1sopzs7",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "InclusiveGateway_1sopzs7",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_0j4pnbs",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_0j4pnbs",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_0j4pnbs",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "InclusiveGateway_1uhxkjz",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "InclusiveGateway_1uhxkjz",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "EndEvent_05zfgny",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "EndEvent_05zfgny",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "EndEvent_05zfgny",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
]
`;
